### ISSUE TITLE
When specifying the full path of rustc on windows, it seems to run rustc incorrectly

### ISSUE THREAD

**Issue by smt923** [AtomLinter/linter-rust#80]
If I specify my full path to rustc.exe like:

![Rustc path](http://i.imgur.com/KSpj1Xn.png)

then whenever I save I get this error:

```
Error: Command failed: C:\Program Files\Rust\bin\rustc.exe --version
'C:\Program' is not recognized as an internal or external command,
operable program or batch file.
    at checkExecSyncError (child_process.js:481:13)
    at Object.execSync (child_process.js:521:13)
    at LinterRust.ableToJSONErrors (file:///C:/Users/admin/.atom/packages/linter-rust/lib/linter-rust.coffee:233:20)
    at LinterRust.ableToJSONErrors (file:///C:/Users/admin/.atom/packages/linter-rust/lib/linter-rust.coffee:1:1)
    at file:///C:/Users/admin/.atom/packages/linter-rust/lib/linter-rust.coffee:33:37
    at LinterRust.lint (file:///C:/Users/admin/.atom/packages/linter-rust/lib/linter-rust.coffee:21:16)
    at Object.lint (file:///C:/Users/admin/.atom/packages/linter-rust/lib/linter-rust.coffee:1:1)
    at promises.push.Promise.then._this.emitter.emit.linter (C:/Users/admin/.atom/packages/linter/lib/linter-registry.js:63:26)
    at C:/Users/admin/.atom/packages/linter/lib/linter-registry.js:62:23
    at Set.forEach (native)
    at LinterRegistry.lint (C:/Users/admin/.atom/packages/linter/lib/linter-registry.js:60:18)
    at file:///C:/Users/admin/.atom/packages/linter/lib/linter.coffee:98:16
    at Function.module.exports.Emitter.simpleDispatch (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\node_modules\event-kit\lib\emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\node_modules\event-kit\lib\emitter.js:125:28)
    at C:/Users/admin/.atom/packages/linter/lib/editor-linter.js:30:20
    at Function.module.exports.Emitter.simpleDispatch (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\node_modules\text-buffer\node_modules\event-kit\lib\emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\node_modules\text-buffer\node_modules\event-kit\lib\emitter.js:129:28)
    at TextBuffer.module.exports.TextBuffer.saveAs (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:1156:27)
    at TextBuffer.module.exports.TextBuffer.save (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:1126:19)
    at TextEditor.module.exports.TextEditor.save (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\src\text-editor.js:755:26)
    at Pane.module.exports.Pane.saveItem (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\src\pane.js:734:18)
    at Pane.saveItem (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\src\pane.js:3:61)
    at Pane.module.exports.Pane.saveActiveItem (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\src\pane.js:717:19)
    at Workspace.module.exports.Workspace.saveActivePaneItem (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\src\workspace.js:695:35)
    at atom-workspace.commandRegistry.add.core:save (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\src\register-default-commands.js:220:32)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\src\command-registry.js:260:29)
    at C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\src\command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:580:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:388:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\src\window-event-handler.js:98:36)
    at HTMLDocument.<anonymous> (C:\Users\admin\AppData\Local\atom\app-1.10.2\resources\app.asar\src\window-event-handler.js:3:61)
```

If I just specify "rustc.exe" it seems to work just fine, thought I'd throw this issue up anyway as I'm not sure what's going on

Edit: this is fixed by enclosing the rustc path in quotes, while cargo works fine - if I then enclose cargo's path in quotes I get this in --dev mode:

```
'""C:\Program' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by White-Oak** [comment]
Thanks for the issue and a good write-up! I'll look closer into it.

**Comment by smt923** [comment]
Just wanted to fix the title/post for correctness, realised I was stupid and that it's nothing to do with --version, it's just not (presumably) stopping at the space in Program Files instead of running the full string as a command

**Comment by Arcanemagus** [comment]
I would _highly_ recommend moving to utilizing `exec()` from [`atom-linter`](https://github.com/steelbrain/atom-linter), it's what essentially all other linters use and we've done our best to handle all the edge cases that have come up so far.

If none of the other parts of `atom-linter` look useful, the `exec()` alone is actually just [`sb-exec`](https://github.com/steelbrain/exec).

This particular issue, and several related ones, are fixed [here](https://github.com/steelbrain/exec/blob/253040324e157876610409a06f653466d0d7efbc/src/index.js#L18-L29).

**Comment by Arcanemagus** [comment]
Also @White-Oak since I noticed @gmist added you to this project directly I've invited you to the AtomLinter team for this project :wink:.

**Comment by White-Oak** [comment]
@smt923 yes, the problem is exactly that, will be fixed :)

@Arcanemagus I remember you recommended it earlier on one of the issues, but I got scared of the lack of the documentation (I am not really a JS developer, so count that in mind). But I looked through it once again and will definitely use it to avoid edge cases such as this one. Thanks for the advice!

And thanks for the invitation. I feel honored :smiley_cat:
